jmxtrans-output/jmxtrans-output-elastic/src/main/java/com/googlecode/jmxtrans/model/output/elastic/ElasticWriter.java

```diff
@@ -0,0 +1,237 @@
+					Index index = new Index.Builder(message).index(indexName).type(TYPE_NAME).build();
+					jestClient.execute(index);
+				} else {
+					log.warn("Unable to submit non-numeric value to Elastic: [{}] from result [{}]", value, result);
+				}
+			}
+		}
+	}
+
+	private String createJsonMessage(Server server, Query query, List<String> typeNames, Result result, Entry<String, Object> values, Object value) throws IOException {
+
+		String keyString = getKeyString(server, query, result, values, typeNames, this.rootPrefix);
+
+		String alias;
+		if (server.getAlias() != null) {
+			alias = server.getAlias();
+		} else {
+			alias = server.getHost() + "_" + server.getPort();
+			alias = StringUtils.cleanupStr(alias);
+		}
+
+		Closer closer = Closer.create();
+		try {
+			ByteArrayOutputStream out = closer.register(new ByteArrayOutputStream());
+			JsonGenerator generator = closer.register(jsonFactory.createGenerator(out, UTF8));
+			generator.writeStartObject();
+			generator.writeStringField("server", alias);
+			generator.writeStringField("metric", keyString);
+			generator.writeNumberField("value", Double.parseDouble(value.toString()));
+			generator.writeStringField("resultAlias", result.getKeyAlias());
+			generator.writeStringField("attributeName", result.getAttributeName());
+			generator.writeStringField("key", values.getKey());
+			generator.writeNumberField("timestamp", result.getEpoch());
+			generator.writeEndObject();
+			generator.close();
+			return out.toString("UTF-8");
+		} catch (Throwable t) {
+			throw closer.rethrow(t);
+		} finally {
+			closer.close();
+		}
+	}
+
+	@VisibleForTesting
+	void setJestClient(JestClient jestClient) {
+		this.jestClient = jestClient;
+	}
+
+	@Override
+	public void validateSetup(Server server, Query query) throws ValidationException {
+		// no validations
+	}
+
+	@Override
+	public void start() throws LifecycleException {
+		super.start();
+
+		if (jestClient == null) {
+			log.info("Create a jest elastic search client for connection url [{}]", connectionUrl);
+			JestClientFactory factory = new JestClientFactory();
+			factory.setHttpClientConfig(
+					new HttpClientConfig.Builder(connectionUrl)
+							.multiThreaded(true)
+							.build());
+			jestClient = factory.getObject();
+
+		}
+		else {
+			log.info("Note: using injected jestClient instead of creating a new one: [{}]", jestClient);
+		}
+
+		createMapping(jestClient, indexName, TYPE_NAME);
+
+	}
+
+	private static void createMapping(JestClient jestClient, String indexName, String typeName) {
+		synchronized (CREATE_MAPPING_LOCK) {
+			try {
+				IndicesExists indicesExists = new IndicesExists.Builder(indexName).build();
+				boolean indexExists = jestClient.execute(indicesExists).isSucceeded();
+
+				if (!indexExists) {
+
+					CreateIndex createIndex = new CreateIndex.Builder(indexName).build();
+					jestClient.execute(createIndex);
+
+					PutMapping putMapping = new PutMapping.Builder(indexName, typeName,
+							"{\n" +
```

Comment:
That big wall of String hurt my eyes. Any chance we could load it from a classpath resource instead ?